common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/RemoteBlockPushResolverSuite.java

```diff
@@ -184,6 +190,10 @@ public void testDeferredBufsAreWrittenDuringOnData() throws IOException {
         new PushBlockStream(TEST_APP, NO_ATTEMPT_ID, 0, 0, 1, 0, 0));
     // This should be deferred
     stream2.onData(stream2.getID(), ByteBuffer.wrap(new byte[3]));
+    verifyMetrics(2, 0, 0, 3, 0, 0);
```

Comment:
I have commented about this earlier in `RemoteBlockPushResolver` - and this will change when it is addressed.